pkg/gatewayserver/io/io.go

```diff
@@ -273,12 +273,13 @@ func (c *Connection) HandleUp(up *ttnpb.UplinkMessage, frontendSync *FrontendClo
 		panic("unreachable")
 	}
 
+	r := receivedAt
+	if _, _, median, _, count := c.RTTStats(100, receivedAt); count > 0 {
+		r = receivedAt.Add(-median / 2)
+	}
+	mdReceivedAtProto := ttnpb.ProtoTimePtr(r)
```

Comment:
```suggestion
	receivedAtGateway := receivedAt
	if _, _, median, _, count := c.RTTStats(100, receivedAt); count > 0 {
		receivedAtGateway = receivedAt.Add(-median / 2)
	}
```

Reply:
Better not declare single character variables in a large scope to avoid confusion. In small scopes, like short loops etc, it's obviously permitted.